File: 1.2Section2/counter.js
function createCounter() {
    let count = 0;  // Local variable in outer scope
    return function increment() {  // Anonymous inner function
        count += 1;  // Implicitly accesses outer scope
        return count;
    };
}

// Test the counter
const counter = createCounter();
console.log(counter());  // Output: 1
console.log(counter());  // Output: 2
console.log(counter());  // Output: 3

File: 1.2Section2/counter.py
def create_counter():
    count = 0  # Local variable in outer scope
    def increment():
        nonlocal count  # Explicitly access outer scope
        count += 1
        return count
    return increment

# Test the counter
counter = create_counter()
print(counter())  # Output: 1
print(counter())  # Output: 2
print(counter())  # Output: 3